doc/flows.puml

```diff
@@ -0,0 +1,222 @@
+' ###############################################################################
+' ########### HERE IS THE LIST OF DIAGRAMS USED FOR THE DOCUMENTATION ###########
+' ###############################################################################
+
+
+' ===========================================================
+' ======================= OAuth2 flow =======================
+' ===========================================================
+
+@startuml
+
+skinparam NoteBackgroundColor #fff
+skinparam NoteBorderColor #000
+
+skinparam sequence {
+	ArrowColor #000
+	ActorBorderColor #000
+
+	LifeLineBorderColor #000
+	LifeLineBackgroundColor #000
+
+	ParticipantBorderColor #000
+	ParticipantBackgroundColor #fff
+	ParticipantFontColor #000
+
+	ActorBackgroundColor #000
+	ActorFontColor #000
+
+        NoteBorderColor #000
+}
+
+skinparam roundcorner 5
+
+actor User                         as user
+participant "Account Web Client"   as account
+database MongoDB                   as mongo
+participant "Connect"              as connect
+
+user -> account : Wants to login
+
+== OAuth2 client initialization ==
+
+account -> connect : Fetch OpenID configuration
+note left
+The **@fwl/oauth2** package mainly provides
+an abstraction for the OAuth2 flow, and a
+secure way to verify JWT for **HS256** and
+**RS256** algorithms
+end note
+connect -> account : Send back OpenID configuration and stores it in-memory
```

Comment:
Everything looks good to me, but I have one question about this part.
I don't know much about this step. Is this the way this application is able to identify what scopes it requests for? Or why do we need this?

Reply:
Yes exactly.
I should have added a link to the package I wrote a few weeks ago documentation: https://github.com/fewlinesco/node-web-libraries/tree/master/packages/oauth2

This package is a small part of what will become the `Account JS SDK`.

I'll add a section about this. Thanks for your feedback!